mobile/lib/modules/asset_viewer/image_providers/cache/image_loader.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_cache_manager/flutter_cache_manager.dart';
+import 'package:immich_mobile/shared/models/store.dart';
+
+/// Loads the codec from the URI and sends the events to the [chunkEvents] stream
+///
+/// Credit to [flutter_cached_network_image](https://github.com/Baseflow/flutter_cached_network_image/blob/develop/cached_network_image/lib/src/image_provider/_image_loader.dart)
+/// for this wonderful implementation of their image loader
+class ImageLoader {
+  static Future<ui.Codec> loadImageFromCache(
+    Uri uri, {
+    required ImageCacheManager cache,
+    required ImageDecoderCallback decode,
+    required StreamController<ImageChunkEvent> chunkEvents,
+    int? height,
+    int? width,
+  }) async {
+    final headers = {
+      'x-immich-user-token': Store.get(StoreKey.accessToken),
+    };
+
+    final stream = cache.getImageFile(
+      uri.toString(),
+      withProgress: true,
+      headers: headers,
+      maxHeight: height,
+      maxWidth: width,
+    );
+
+    await for (final result in stream) {
+      if (result is DownloadProgress) {
+        // We are downloading the file, so update the [chunkEvents]
+        chunkEvents.add(
+          ImageChunkEvent(
```

Comment:
Do we actually need these events? Afaik, we don't use them anywhere to show download progress? The loading spinners don't need the information

Reply:
We don't use it anywhere, and we wouldn't ever want to with thumbnails. With "full size" remote images? We may want to support this, but we do multi-stage loading... Hmm, I'm leaning towards tearing this out, or maybe making it optional if `chunkEvents` is sent in.

Reply:
I made it optional and currently only `ImmichRemoteImageProvider` uses it. The gallery pager does use it, so we could optionally show some loading indicator in our placeholder for that.